### ISSUE TITLE
Engine tree closed on infra failures (networking issues)

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#101724]
Many of the bots are closed right now on infra failures.

For example Linux Fuchsia is closed on `Upload Fuchsia Artifacts`:
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia/13355/overview

Linux Arm Host Engine is closed on `Checkout source code`:
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia/13355/overview

(This may be related to the networking issues related in https://github.com/flutter/flutter/issues/101719)

**Comment by gaaclarke** [comment]
I looked at the logs a bit more and it seems like the `Checkout source code` failures are definitely the same `gclient sync` problem as https://github.com/flutter/flutter/issues/101719

@zanderso thinks the fuchsia failure is because we are trying to upload an artifact that isn't present

**Comment by zanderso** [comment]
@yusuf-goog

**Comment by yusuf-goog** [comment]
cc @CaseyHillers in reference to #101719

**Comment by yusuf-goog** [comment]
cc @XilaiZhang if this is a frob issue

**Comment by gaaclarke** [comment]
I ran a `gclient sync` locally and it ran fine

**Comment by gaaclarke** [comment]
I don't think this is a frob issue.  The gclient sync problems shouldn't have any relation to frob.

**Comment by CaseyHillers** [comment]
This isn't a FRoB issue.

**Comment by gaaclarke** [comment]
Sorry, the frob confusion was my fault, I pasted the wrong url in the description.  This issue seems to be the same issue as https://github.com/flutter/flutter/issues/101718. I update the description.

**Comment by CaseyHillers** [comment]
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia/13355/overview is a timeout (at 60 minutes 3 seconds it was killed because of a 60 minute timeout).

The issue is in https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Arm%20Host%20Engine/6435/overview

```
Error: Command 'git -c core.deltaBaseCacheLimit=2g fetch origin d40d15e994ed60d32bcfc9ab87004dfb028dfbd6 --verbose --no-tags' returned non-zero exit status 128 in /b/s/w/ir/cache/builder/src/third_party/harfbuzz
```


@yusuf-goog can you check that our mirror of harfbuzz is working?

**Comment by yusuf-goog** [comment]
I can clone and checkout the commit in https://flutter.googlesource.com/mirrors/engine/+/refs/heads/main/DEPS#131 from https://flutter.googlesource.com/third_party/harfbuzz

**Comment by godofredoc** [comment]
Could this be related to quota and non authenticated requests?

**Comment by yusuf-goog** [comment]
Looks like there might be a gob event going on. Filed a bug: https://b.corp.google.com/issues/228889035

**Comment by yusuf-goog** [comment]
FYI:

@Stuart Morgan could reproduce bad clones locally

He saw:
```
$ git clone https://flutter.googlesource.com/third_party/fontconfig
Cloning into 'fontconfig'...
warning: You appear to have cloned an empty repository.
```

When it is definitely not an empty repo

**Comment by yusuf-goog** [comment]
gob team confirms that the empty repo could be due to permission issues. They are looking into it.

**Comment by yusuf-goog** [comment]
A permission change to the repo was reverted (https://flutter-review.googlesource.com/c/Public-Projects/+/28943) and seems like it caused the inability to clone/fetch.

**Comment by yusuf-goog** [comment]
Retries are running now.

**Comment by yusuf-goog** [comment]
Everything seems to be up and running now. Closing. Please reopen if issues persist.
